app/Http/Controllers/ReportCrudController.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Transaction;
+use App\Services\UserService;
+use Backpack\CRUD\app\Http\Controllers\CrudController;
+use Backpack\CRUD\app\Library\CrudPanel\CrudPanel;
+use Backpack\CRUD\app\Library\CrudPanel\CrudPanelFacade as CRUD;
+
+/**
+ * Class SubjectCategoryCrudController
+ *
+ * @property-read CrudPanel $crud
+ */
+class ReportCrudController extends CrudController
+{
+    use \Backpack\CRUD\app\Http\Controllers\Operations\CreateOperation;
+    use \Backpack\CRUD\app\Http\Controllers\Operations\DeleteOperation;
+    use \Backpack\CRUD\app\Http\Controllers\Operations\ListOperation;
+    use \Backpack\CRUD\app\Http\Controllers\Operations\UpdateOperation;
+    use \Backpack\Pro\Http\Controllers\Operations\InlineCreateOperation;
+    
+    public function setup(): void
+    {
+        CRUD::setModel(\App\Models\Transaction::class); // Transaction
+        CRUD::setRoute(config('backpack.base.route_prefix').'/report');
+        CRUD::setEntityNameStrings('transaction', 'transactions');
+
+        if (!UserService::hasAccessTo('bookkeeping')) {
+            $this->crud->denyAllAccess();
+        }
+    }
+
+    protected function setupListOperation(): void
+    {
+        CRUD::column('name');
+        CRUD::column('description');
+    }
+
+    protected function setupCreateOperation(): void
+    {
+        CRUD::setValidation(\App\Http\Requests\ReportRequest::class);
+
+        CRUD::field('name');
+        CRUD::field('description');
+    }
+
+    protected function setupUpdateOperation(): void
+    {
+        $this->setupCreateOperation();
+    }
+    
+    public function showReport()
+    {
+        $incomeCategories = [1, 3, 4];
+        $expenseCategories = [2, 5, 6, 7, 8];
+    
+        $incomes = \DB::table('transactions')
+            ->whereIn('transaction_category_id', $incomeCategories)
+            ->groupBy('transaction_category_id')
+            ->select(\DB::raw('transaction_category_id, SUM(amount) as total'))
+            ->pluck('total', 'transaction_category_id')
+            ->toArray();
```

Comment:
While the queries here do their job, I would recommend you to use Laravel Eloquent. It is a very powerful tool and can simplify the code. On top of that, working with Laravel's Models you make use of the getters, which means Laravel will convert the properties to the right type.

In this case, the problem was that `DB::table...` was returning the sum as string, and the javascript library needs integers to display info. If it is a string, then it is not a number, even if it looks like `["1", "2", "3", "4"]`. The library expected instead `[1,2,3,4]`.

A workaround your query would be to use a map function and cast the values (sum) to _int_, like so:

```
$incomes = \DB::table('transactions')
            ->whereIn('transaction_category_id', $incomeCategories)
            ->groupBy('transaction_category_id')
            ->select(\DB::raw('transaction_category_id, SUM(amount) as total'))
            ->pluck('total', 'transaction_category_id')
            ->map(fn ($value) => (int) $value) // <----- this line here fixes the issue
            ->toArray();
```

BUT, I would recommend you to re-write this queries using the Model Transaction:

```
Transaction::query()
   ->where(...)
   ->sum('column_name') // <---- Laravel Eloquent has a helper that will do all the magic for you, and it will return an int!
```

Please, let me know if you need help with this.